spec/controllers/concerns/idv/threat_metrix_concern_spec.rb

```diff
@@ -0,0 +1,86 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe Idv::ThreatMetrixConcern, type: :controller do
+  controller ApplicationController do
+    include Idv::ThreatMetrixConcern
+
+    before_action :override_csp_for_threat_metrix
+
+    def index; end
+  end
+
+  describe '#override_csp_for_threat_metrix' do
+    context 'non-production environment' do
+      before do
+        allow(Rails.env).to receive(:production?).and_return(false)
+      end
+      it 'does not modify CSP headers for SSN step' do
+        assert_csp_is_not_modified 'ssn'
+      end
+      it 'does not modify CSP headers for any other step' do
+        assert_csp_is_not_modified 'some_other_step'
+      end
+    end
+
+    context 'production enviroment' do
+      before do
+        allow(Rails.env).to receive(:production?).and_return(true)
+      end
+
+      context 'ff is set' do
+        before do
+          allow(IdentityConfig.store).to receive(:proofing_device_profiling_collecting_enabled).
+            and_return(true)
+        end
+        it 'modifies CSP headers for SSN step' do
+          assert_csp_is_modified 'ssn'
+        end
+
+        it 'does not modify CSP headers for any other step' do
+          assert_csp_is_not_modified 'some_other_step'
+        end
+      end
+
+      context 'ff is not set' do
+        before do
+          allow(IdentityConfig.store).to receive(:proofing_device_profiling_collecting_enabled).
+            and_return(false)
+        end
+
```

Comment:
another option would be to move this `before` block up to before all specs, and then have it use a `let` variable to override?

```suggestion
        before do
          allow(IdentityConfig.store).to receive(:proofing_device_profiling_collecting_enabled).
            and_return(proofing_device_profiling_collecting_enabled)
        end
        
        let(:proofing_device_profiling_collecting_enabled) { false )
```

Reply:
I like this

Reply:
By which I mean, "I will implement this suggestion"